src/main/java/org/bookkeepervisualmanager/config/ConfigurationStore.java

```diff
@@ -0,0 +1,38 @@
+/*
+ Licensed to Diennea S.r.l. under one
+ or more contributor license agreements. See the NOTICE file
+ distributed with this work for additional information
+ regarding copyright ownership. Diennea S.r.l. licenses this file
+ to you under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance
+ with the License.  You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing,
+ software distributed under the License is distributed on an
+ "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ KIND, either express or implied.  See the License for the
+ specific language governing permissions and limitations
+ under the License.
+
+ */
+package org.bookkeepervisualmanager.config;
+
+import java.util.function.BiConsumer;
+
+/**
+ * Stores configuration
+ *
+ * @author matteo.minardi
+ */
+public interface ConfigurationStore extends AutoCloseable {
```

Comment:
Honestly I think is quite overkilling.
We only need a minimal logic to read a file and/or use only metadataServiceURi
We won't create multiple implementations of this interface.
I see you took Carapace Proxy as an example, but in that case we have at least two complete and production ready  implementations (on file and on database)